client/state/experiments/hooks.ts

```diff
@@ -1,60 +1,68 @@
 /**
  * External Dependencies
  */
-import { useSelector } from 'react-redux';
+import { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
 
 /**
  * Internal Dependencies
  */
-import { getVariationForUser, isLoading, getAnonId, nextRefresh } from './selectors';
+import { fetchExperiments } from 'calypso/state/data-layer/wpcom/experiments';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { getVariationForUser, isLoading, nextRefresh } from './selectors';
 
 /**
- * Returns true if the variations are loading for the current user
+ * Keeps track of whether an event has fired to generate an anonId
+ * Internal state not to be used anywhere else
  */
-export function useIsLoading(): boolean {
-	return useSelector( isLoading );
-}
+let hasAnonIdGeneratingTracksEventFired = false;
 
 /**
- * Returns the user's assigned variation for a given experiment
+ * Gets (and initialises if not yet initialised) a users assignment to an experiment.
  *
- * @param experiment The name of the experiment
- */
-export function useVariationForUser( experiment: string ): string | null {
-	return useSelector( ( state ) => getVariationForUser( state, experiment ) );
-}
-
-/**
- * Gets the anon id for the user, if set
- */
-export function useAnonId(): string | null {
-	return useSelector( getAnonId );
-}
-
-/**
- * Get the time for the next variation refresh
+ * Usage tips:
+ * - Account for loading state as assignment is now server-side.
+ * - Account for null variations: Provide the fallback experience.
+ *
+ * @param experimentName The name of the experiment
+ *
+ * @returns [isVariationLoading, variation]
  */
-export function useNextRefresh(): number {
-	return useSelector( nextRefresh );
-}
+export function useExperiment( experimentName: string ): [ boolean, string | null ] {
+	const dispatch = useDispatch();
 
-type ExperimentInfo = {
-	variation: string | null;
-	anonId: string | null;
-	nextRefresh: number;
-	isLoading: boolean;
-};
+	if ( ! experimentName ) {
```

Comment:
Do we need this check given that `experimentName`'s type is `string`?

Reply:
It is possible for plain Javascript to be accessing this too, and we should be checking for empty strings anyway.